### PR TITLE
Enable opt-in junit5 parallel tests

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR enables junit5 parallel tests for profile `parallel-tests`. Note that this is opt-in: by default, all tests within a class run sequentially. However, you can now annotate these classes with the `@Execution` annotation to opt-in and run your tests in parallel. Default configuration is to run up to half of the number of cores in parallel.

As such, this PR is more of a stepping stone to allow further options for parallel test execution.

## Related issues

related #5873

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by npepinpe** [comment]
Note that this is based on https://github.com/zeebe-io/zeebe/pull/5937 so it includes some commits from it. Hold off from merging until that one is merged.

**Comment by npepinpe** [comment]
@korthout ready for review :slightly_smiling_face:

**Comment by npepinpe** [comment]
Actually scratch that, sorry - I'll ping you again when it's ready

**Comment by npepinpe** [comment]
Now it's ready for review :slightly_smiling_face: 

The one build which failed was due to address in use, which should be fixed once it's merged in develop as another PR was merged since then.

One small note, we now have a lot of shared config between surefire and failsafe, but I'm not aware of a way to share config between plugins? If you do I'd be happy to know it :+1:

**Comment by korthout** [comment]
>a way to share config between plugins

I don't think that's possible with maven. Maybe in the future it allows to define general re-usable definitions or so, but the best thing that would be available now is properties, which to my knowledge don't allow nested structures.

File: parent/pom.xml
Comment (korthout) [inline parent/pom.xml:1270]: So this will take the value 6 in our ci? I think we should set this explicitly in the scripts, just like the `forkCount`. It helps us reason about this for each specific case

**Review by korthout — COMMENTED** [review]
Nice work! I can definitely see a benefit to this for specific cases. The only thing I'm wondering about is how this relates to the jvm forks provided by surefire/failsafe plugin. Should I read this `junitThreadCount` as the number of threads on 1 jvm to run tests concurrently (i.e. not really in parallel)?

**Comment by npepinpe** [comment]
Exactly. junit5 will run tests in a thread pool, so that's another dimension - you have stages > maven parallel build > forks > junit5 threads. To be honest, I'm simply enabling it so it's available - I imagine unit tests would benefit the most, and integration tests not so much.

**Review by korthout — APPROVED** [review]
LGTM 🍪

**Comment by npepinpe** [comment]
Yeah, I guess we need to wait until the other backports are through before this, otherwise it's going to be a mess.

**Comment by korthout** [comment]
@npepinpe It seems you skipped the stable/0.24 backport. Can you please add this or was it purposely omitted?

**Comment by npepinpe** [comment]
I'm waiting for https://github.com/zeebe-io/zeebe/pull/5961 to be merged :slightly_smiling_face:

**Comment by korthout** [comment]
@npepinpe #5963 has been merged (I think you meant that one), please remember the 0.24 backport.

**Comment by npepinpe** [comment]
No, the one you linked is for 0.25, and we're talking about 0.24 here, no?

**Pull request merged**
